src/app/codegen-data-model-provider/CodegenDataModelProvider.cpp

```diff
@@ -516,6 +628,44 @@ DataModel::CommandEntry CodegenDataModelProvider::NextAcceptedCommand(const Conc
 
 std::optional<DataModel::CommandInfo> CodegenDataModelProvider::GetAcceptedCommandInfo(const ConcreteCommandPath & path)
 {
+    // Command handler interface MAY override lists of commands
+    CommandHandlerInterface * handler_interface =
+        CommandHandlerInterfaceRegistry::Instance().GetCommandHandler(path.mEndpointId, path.mClusterId);
+
+    if (handler_interface != nullptr)
+    {
+        struct Context
+        {
+            bool commandExists;
+            CommandId targetCommand;
+        } context{ false, path.mCommandId };
+
+        // if a list is available, try to see if the command exists
+        CHIP_ERROR err = handler_interface->EnumerateAcceptedCommands(
+            path,
+            [](CommandId command, void * closure) -> Loop {
+                auto * ctx = static_cast<Context *>(closure);
+                if (ctx->targetCommand == command)
+                {
+                    ctx->commandExists = true;
+                    return Loop::Break;
+                }
+                return Loop::Continue;
+            },
+            &context);
+
+        if (err != CHIP_ERROR_NOT_IMPLEMENTED)
+        {
+            // TODO: we know the command exists, but don't actually have info such as flags. Ideally CHI should
```

Comment:
See above.  If there is no ZAP metadata here the API implementation is doing it wrong.  We should be using CommandEntryFrom.

Also, this is what FindExact was added for, I guess, but we're not using it?  Why is it there, then?

Reply:
I started writing this code and then figured out a common class is better ... implemented it and did not use it. Updated now.